packages/cmake/package.py

```diff
@@ -0,0 +1,207 @@
+    depends_on('libuv@1.10.0:',  when='@3.12.0:~ownlibs')
+    depends_on('rhash',          when='@3.8.0:~ownlibs')
+    depends_on('qt',             when='+qt')
+    depends_on('python@2.7.11:', when='+doc', type='build')
+    depends_on('py-sphinx',      when='+doc', type='build')
+    depends_on('openssl', when='+openssl')
+    depends_on('openssl@:1.0.99', when='@:3.6.9+openssl')
+    depends_on('ncurses',        when='+ncurses')
+
+    # Cannot build with Intel, should be fixed in 3.6.2
+    # https://gitlab.kitware.com/cmake/cmake/issues/16226
+    patch('intel-c-gnu11.patch', when='@3.6.0:3.6.1')
+
+    # https://gitlab.kitware.com/cmake/cmake/issues/18232
+    patch('nag-response-files.patch', when='@3.7:3.12')
+
+    conflicts('+qt', when='^qt@5.4.0')  # qt-5.4.0 has broken CMake modules
+
+    # https://gitlab.kitware.com/cmake/cmake/issues/18166
+    conflicts('%intel', when='@3.11.0:3.11.4')
+    conflicts('%intel@:14', when='@3.14:',
+              msg="Intel 14 has immature C++11 support")
+
+    sanity_check_is_file = ['bin/cmake']
+
+    phases = ['bootstrap', 'build', 'install']
+
+    def setup_build_environment(self, env):
+        super(Cmake, self).setup_build_environment(env)
+        env.unset('DASHBOARD_TEST_FROM_CTEST')
+
+    def flag_handler(self, name, flags):
+        if name == 'cxxflags' and self.compiler.name == 'fj':
+            cxx11plus_flags = (self.compiler.cxx11_flag,
+                               self.compiler.cxx14_flag)
+            cxxpre11_flags = (self.compiler.cxx98_flag)
+            if any(f in flags for f in cxxpre11_flags):
+                raise ValueError('cannot build cmake pre-c++11 standard')
+            elif not any(f in flags for f in cxx11plus_flags):
+                flags.append(self.compiler.cxx11_flag)
+        return (flags, None, None)
+
+    def bootstrap_args(self):
+        spec = self.spec
+        args = [
+            '--prefix={0}'.format(self.prefix),
+            '--parallel={0}'.format(make_jobs)
+        ]
+
+        if '+ownlibs' in spec:
+            # Build and link to the CMake-provided third-party libraries
+            args.append('--no-system-libs')
+        else:
+            # Build and link to the Spack-installed third-party libraries
+            args.append('--system-libs')
+
+            if spec.satisfies('@3.2:'):
+                # jsoncpp requires CMake to build
+                # use CMake-provided library to avoid circular dependency
+                args.append('--no-system-jsoncpp')
+
+        if '+qt' in spec:
+            args.append('--qt-gui')
+        else:
+            args.append('--no-qt-gui')
+
+        if '+doc' in spec:
+            args.append('--sphinx-html')
+            args.append('--sphinx-man')
+
+        # Now for CMake arguments to pass after the initial bootstrap
+        args.append('--')
+
+        # Make sure to create an optimized release build
+        args.append('-DCMAKE_BUILD_TYPE=Release')
+
```

Comment:
```diff
+ # Install CMake correctly, even if the `spack install` runs inside a ctest environment
+ args.append('-DCMake_TEST_INSTALL=OFF')
+
```

This also works (I tested it) instead of unsetting the `$DASHBOARD_TEST_FROM_CTEST` (do we know what other effects this may have?). See https://gitlab.kitware.com/cmake/cmake/blob/master/Tests/CMakeInstall.cmake#L1-6.

Reply:
1. I think, unsetting `$DASHBOARD_TEST_FROM_CTEST` shouldn't hurt. As in "get the build environment as reproducible as possible".
2. Adding the `-DCMake_TEST_INSTALL=OFF` looks also good.

What about doing both?

Reply:
All fine with me, I will also merge it how it is now. I leave it to your decision.